From 4ce6f485207da9404c5cc0aa90d0ccd745096294 Mon Sep 17 00:00:00 2001
From: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Date: Fri, 3 Dec 2021 18:15:52 +0800
Subject: [PATCH] arm64: fix abi change caused by ILP32
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 4ce6f485207da9404c5cc0aa90d0ccd745096294


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I4KCU2
CVE: NA

----------------------------------------

One of the ILP32 patchset rename 'compat_user_mode' and
'compat_thumb_mode' to 'a32_user_mode' and 'a32_user_mode'. But these
two macros are used in some opensource userspace application. To keep
compatibility, we redefine these two macros.

Fixes: 23b2f00 ("arm64: rename functions that reference compat term")
Signed-off-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Reviewed-by: liwei <liwei391@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/include/asm/ptrace.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/include/asm/ptrace.h b/arch/arm64/include/asm/ptrace.h
index 34ed891da81b..9193e40b0cce 100644
--- a/arch/arm64/include/asm/ptrace.h
+++ b/arch/arm64/include/asm/ptrace.h
@@ -220,6 +220,8 @@ static inline void forget_syscall(struct pt_regs *regs)
 #define a32_thumb_mode(regs) (0)
 #endif
 
+#define compat_thumb_mode(regs) a32_thumb_mode(regs)
+
 #define user_mode(regs)	\
 	(((regs)->pstate & PSR_MODE_MASK) == PSR_MODE_EL0t)
 
@@ -227,6 +229,8 @@ static inline void forget_syscall(struct pt_regs *regs)
 	(((regs)->pstate & (PSR_MODE32_BIT | PSR_MODE_MASK)) == \
 	 (PSR_MODE32_BIT | PSR_MODE_EL0t))
 
+#define compat_user_mode(regs) a32_user_mode(regs)
+
 #define processor_mode(regs) \
 	((regs)->pstate & PSR_MODE_MASK)
 
-- 
2.26.2

